tracer/src/Datadog.Trace/Util/ProcessHelpers.cs

```diff
@@ -155,112 +164,21 @@ public static void GetCurrentProcessInformation(out string processName, out stri
-                                processInfo = StartWithDoNotTrace(processStartInfo, command.DoNotTrace);
-                                break;
-                            }
-                        }
-                    }
-                }
-                else if (!string.Equals(command.Cmd, "where", StringComparison.OrdinalIgnoreCase))
-                {
-                    var cmdResponse = await RunCommandAsync(new Command("where", processStartInfo.FileName)).ConfigureAwait(false);
-                    if (cmdResponse?.ExitCode == 0 &&
-                        cmdResponse.Output.Split(["\n", "\r\n"], StringSplitOptions.RemoveEmptyEntries) is { Length: > 0 } outputLines &&
-                        outputLines[0] is { Length: > 0 } processPath &&
-                        File.Exists(processPath))
-                    {
-                        processStartInfo.FileName = processPath;
-                        processInfo = StartWithDoNotTrace(processStartInfo, command.DoNotTrace);
-                    }
-                }
-                else
-                {
-                    throw;
-                }
-            }
-
-            if (processInfo is null)
-            {
-                return null;
-            }
-
-            using var disposableProcessInfo = processInfo;
-            if (input is not null)
-            {
-                await processInfo.StandardInput.WriteAsync(input).ConfigureAwait(false);
-                await processInfo.StandardInput.FlushAsync().ConfigureAwait(false);
-                processInfo.StandardInput.Close();
-            }
-
-            var outputStringBuilder = new StringBuilder();
-            var errorStringBuilder = new StringBuilder();
-            while (!processInfo.HasExited)
-            {
-                if (!processStartInfo.UseShellExecute)
-                {
-                    outputStringBuilder.Append(await processInfo.StandardOutput.ReadToEndAsync().ConfigureAwait(false));
-                    errorStringBuilder.Append(await processInfo.StandardError.ReadToEndAsync().ConfigureAwait(false));
-                }
-
-                await Task.Delay(15).ConfigureAwait(false);
+                Log.Warning("Process timed out after {Timeout}. Killing process.", command.Timeout);
+                processInfo.Kill();
+                return new CommandOutput(outputStringBuilder.ToString(), errorStringBuilder.ToString(), exitCode: -1, timedOut: true);
             }
 
             if (!processStartInfo.UseShellExecute)
             {
-                outputStringBuilder.Append(await processInfo.StandardOutput.ReadToEndAsync().ConfigureAwait(false));
-                errorStringBuilder.Append(await processInfo.StandardError.ReadToEndAsync().ConfigureAwait(false));
+                outputStringBuilder.Append(processInfo.StandardOutput.ReadToEnd());
+                errorStringBuilder.Append(processInfo.StandardError.ReadToEnd());
```

Comment:
If the output is redirected then you should have a background thread that pumps the stdout/stderr (we have a helper for that in the integration tests). If the command outputs a lot of text, it will fill the buffer and writing to the console will become a blocking operation for the target application, leading to a deadlock (we wait for the app to exit, the app can't complete because it's waiting for somebody to read the output)